getcontext/tracing/trace.py

```diff
@@ -1,7 +1,45 @@
+from langsmith.run_trees import RunTree
 
-class Trace:
-    HACKY_CONSTANT = None
-    
-    def __init__(self, result, run_tree):
+
+class Trace:    
+    def __init__(self, result, run_tree: RunTree):
         self.result = result
         self.run_tree = run_tree
+    
+    # TODO: used typed stuff from sdk here
+    def add_evaluator(self, span_name: str, evaluator: dict):        
+        if span_name == self.run_tree.name:
+            raise ValueError("Cannot add evaluator to unit test function.")
+
+        langsmith_run = self._find_run(span_name)
+        if langsmith_run.extra is None:
+            langsmith_run.extra = {}
+        context_ai_options = langsmith_run.extra.setdefault("context_ai_options", {})
```

Comment:
Consider using a constant for this magic string? That way we can add a comment explaining that this is a 'known string' that will be interpreted server side.